### PR TITLE
Fix to prevent: "log collection failed: circuit breaking exception"

### PR DESCRIPTION
**Pull request by JohanKJSchreurs**
# Fix to prevent: "log collection failed: circuit breaking exception"

Part of implementation to fix issue #170

##  Add log_level parameter to get_log_entries.

Setting log_level should help to limit the amount of memory used in Elasticsearch while retrieving logs.
The cause of the circuit_breaker_exception is that amount of memory being used to query the logs, is getting too close to the maximum limit allowed.

This circuit breaker is apparently a mechanism to prevent that the whole Elasticsearch process runs out of memory which would make it crash. Therefore the request is cut off before it really exhausts memory.

There seem to be either too many log records, or there may be fields that contain a large amount of data (long messages perhaps).

## Multiple code bases involved

This has to be implemented throughout three repositories each with there respective PR.
This PR is the main PR (because the issue was logged on the openeo-python-driver)

- https://github.com/Open-EO/openeo-python-driver : 
    * The main PR.
    * The Python driver needs to link through log_level parameter to backend implementation in GeoPySpark driver.
- https://github.com/Open-EO/openeo-geopyspark-driver :
    *  Where logs are actually retrieved from Elasticsearch.
    *  PR: https://github.com/Open-EO/openeo-geopyspark-driver/pull/382
- https://github.com/Open-EO/openeo-python-client  
    * needs to be use log_level when retrieving logs via the openeo-python-driver
    * PR: https://github.com/Open-EO/openeo-python-client/pull/408

### CODE REVIEW

File: openeo_driver/backend.py [hunk 1]
```diff
@@ -439,7 +439,13 @@ def get_results(self, job_id: str, user_id: str) -> Dict[str, dict]:
         # TODO: eliminate this method in favor of `get_result_assets`
         raise NotImplementedError
 
-    def get_log_entries(self, job_id: str, user_id: str, offset: Optional[str] = None) -> Iterable[dict]:
+    def get_log_entries(
+        self,
+        job_id: str,
+        user_id: str,
+        offset: Optional[str] = None,
+        log_level: Optional[str] = None,
```
Comment (soxofaan) [inline openeo_driver/backend.py:447]: FYI: request parameter will be `level` iso `log_level` (https://github.com/Open-EO/openeo-api/pull/488) 
it's probably better to stick to that convention everywhere

File: openeo_driver/views.py [hunk 2]
```diff
@@ -1258,9 +1258,12 @@ def download_job_result_signed(job_id, user_base64, secure_key, filename):
     @blueprint.route('/jobs/<job_id>/logs', methods=['GET'])
     @auth_handler.requires_bearer_auth
     def get_job_logs(job_id, user: User):
-        offset = request.args.get('offset')
+        offset = request.args.get("offset")
+        log_level = request.args.get("log_level")
```
Comment (soxofaan) [inline openeo_driver/views.py:1262]: as noted above : `log_level` -> `level`

**Review by soxofaan — COMMENTED** [review]
apart from `log_level` -> `level`, this looks ready for merging

File: openeo_driver/backend.py (hunk 1 above)
Comment (JohanKJSchreurs) [reply openeo_driver/backend.py]: Done.

File: openeo_driver/dummy/dummy_backend.py [hunk 3]
```diff
@@ -731,11 +732,21 @@ def get_result_assets(self, job_id: str, user_id: str) -> Dict[str, dict]:
...
+        user_id: str,
+        offset: Optional[str] = None,
+        level: Optional[str] = None,
+    ) -> Iterable[dict]:
         self._get_job_info(job_id=job_id, user_id=user_id)
         default_logs = [{"id": "1", "level": "info", "message": "hello world"}]
         for log in self._custom_job_logs.get(job_id, default_logs):
             if isinstance(log, dict):
+                actual_level = normalize_log_level(log.get("log_level"))
+                requested_level = normalize_log_level(level)
```
Comment (soxofaan) [inline openeo_driver/dummy/dummy_backend.py:747]: `requested_level` can be moved outside of for loop I think

**Review by soxofaan — COMMENTED** [review]
just minor note, looks ready for merge

I would also bump the minor version level in `_version.py` to mark this as an API change (which can be depended on in geopyspark driver)

File: openeo_driver/dummy/dummy_backend.py (hunk 3 above)
Comment (JohanKJSchreurs) [reply openeo_driver/dummy/dummy_backend.py]: Indeed, no need to repeat that function call.  Done.
Minor version bumped as well.

**Pull request merged**